app/models/user.rb

```diff
@@ -8,8 +10,9 @@ class User < ApplicationRecord
     format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
   validates :password, presence: true,
     length: {minimum: Settings.user_pass_length_min}, allow_nil: true
-  before_save{email.downcase!}
-  has_secure_password
+    before_save{email.downcase!}
```

Comment:
tạo 1 method private nhé

Reply:
tạo xong rồi em phải gọi lại vào đây chứ?
```suggestion
    before_save :downcase_email
```